### PR TITLE
[RISC-V] R2RDump to handle System.Private.CoreLib

### PR DESCRIPTION
**Pull request by tomeksowi**
Now that we have a crossgen2'ed System.Private.CoreLib.dll, adjust R2RDump to handle it.

Part of #84834
cc @wscho77 @HJLeee @clamp03 @JongHeonChoi @t-mustafin @gbalykov @viewizard @ashaurtaev @brucehoult @sirntar @yurai007 @Bajtazar @bartlomiejko @rzsc

### CODE REVIEW

**Review by sirntar — APPROVED** [review]

**Review by yurai007 — APPROVED** [review]

File: src/coreclr/tools/r2rdump/CoreDisTools.cs [hunk 1]
```diff
@@ -29,9 +29,6 @@ public enum TargetArch
         [DllImport(_dll, CallingConvention = CallingConvention.Cdecl)]
         public static extern IntPtr InitBufferedDisasm(TargetArch Target);
 
-        [DllImport(_dll, CallingConvention = CallingConvention.Cdecl)]
```
Comment (ivdiazsa) [inline src/coreclr/tools/r2rdump/CoreDisTools.cs:32]: I've a question here. Why are we removing this? Is it no longer used anywhere?

Comment (tomeksowi) [reply src/coreclr/tools/r2rdump/CoreDisTools.cs:32]: [Yes](https://github.com/search?q=repo%3Adotnet%2Fruntime+DumpCodeBlock+language%3AC%23&type=code&l=C%23)

Comment (ivdiazsa) [reply src/coreclr/tools/r2rdump/CoreDisTools.cs:32]: Ok I see. I kicked off an outerloop run just to be safe this entire PR is good to go. If it returns all green, then we can approve and merge this.

**Review by ivdiazsa — APPROVED** [review]

**Comment by ivdiazsa** [comment]
All comments have been addressed and the CI came back fully clean, so merging this now.

**Pull request merged**